src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThreadGroup.java

```diff
@@ -174,16 +174,13 @@ private Object readResolve() {
         execution = CpsFlowExecution.PROGRAM_STATE_SERIALIZATION.get();
         setupTransients();
         assert execution!=null;
-        if (scripts != null) { // compatibility: the field will be null in old programs
+        if (scripts != null && !scripts.isEmpty()) { // compatibility: the field will be null in old programs
```

Comment:
```suggestion
        if (/* compatibility: the field will be null in old programs */ scripts != null && !scripts.isEmpty()) {
```

to clarify which part of the clause the comment applies to

Reply:
addressed in f15b02ddf3268eac1df80d25f14745975d68baa4